File: src/main/java/de/arrobait/antlers/psi/AntlersNamedElement.java
package de.arrobait.antlers.psi;

import com.intellij.psi.PsiElement;
import com.intellij.psi.PsiNameIdentifierOwner;
import org.jetbrains.annotations.NotNull;

public interface AntlersNamedElement extends PsiNameIdentifierOwner {
    @NotNull
    String getName();

    @NotNull
    PsiElement getId();
}

File: src/main/java/de/arrobait/antlers/psi/AntlersNamedElement.kt
package de.arrobait.antlers.psi

import com.intellij.psi.PsiElement
import com.intellij.psi.PsiNameIdentifierOwner

interface AntlersNamedElement : AntlersPsiElement, PsiNameIdentifierOwner {
    override fun getName(): String

    fun getId(): PsiElement?
}
